### ISSUE TITLE
Permission issue with logrotate

### ISSUE THREAD

**Issue by samueljon** [netdata/netdata#2794]
**Operating System:** Red Hat flavor - Centos 7.x - 64bit
**Installation:** via `bash <(curl -Ss https://my-netdata.io/kickstart-static64.sh)` 
**Installation location:** /opt
**SeLinux:** enabled

**Issue description:** 

I'm getting a logrotate permission error on all my CentOS based installations when using the prebuilt-static-binaries ( although i'm not certain that it's related but I do not recall seeing this on source builded versions ). 

```log
/etc/cron.daily/logrotate:

error: error opening /opt/netdata/var/log/netdata/error.log: Permission denied
```

I've tried the following without luck:

1. Changing the permission on the `/opt/netdata/var/log/netdata` folder to `chmod 775`
2. Adding `su` to use the netdata's user and group when rotating
```
/opt/netdata/var/log/netdata/*.log {
	su netdata netdata
	daily
	missingok
	rotate 14
	compress
	delaycompress
	notifempty
	sharedscripts
	postrotate
		/bin/kill -HUP `pidof netdata 2>/dev/null` 2>/dev/null || true
	endscript
}
```

Is it possible that this a selinux issue ? 
Perhaps the selinux context is not properly set and could be set to var_log_t or netdata security contect that allows logging 
```
semanage fcontext -a -t var_log_t "<directory>(/.*)?"
restorecon -v <directory> 
```
The Selinux security contexts are as follows compared to `/var/log/messages` 

```bash
[root@machine log]# ls -laZ messages
-rw-------. root root system_u:object_r:var_log_t:s0   messages
[root@machine log]# ls -laZ /opt/netdata/var/log/netdata/error.log
-rw-rw----. netdata netdata system_u:object_r:usr_t:s0       /opt/netdata/var/log/netdata/error.log
[root@machine log]#
```

**Comment by samueljon** [comment]
Example of log folder contents and it's permissions

```bash
[root@machine netdata]# pwd
/opt/netdata/var/log/netdata
[root@machine netdata]# ls -alZ
drwxrwxr-x. netdata root    system_u:object_r:usr_t:s0       .
drwxrwxr-x. root    root    unconfined_u:object_r:usr_t:s0   ..
-rw-rw----. netdata netdata system_u:object_r:usr_t:s0       access.log
-rw-r--r--. netdata netdata system_u:object_r:usr_t:s0       access.log.1
-rw-r--r--. netdata netdata system_u:object_r:usr_t:s0       debug.log
-rw-rw----. netdata netdata system_u:object_r:usr_t:s0       error.log
-rw-rw----. netdata netdata system_u:object_r:usr_t:s0       error.log.1
-rw-r--r--. netdata netdata system_u:object_r:usr_t:s0       error.log.2.gz
-rw-r--r--. netdata netdata system_u:object_r:usr_t:s0       .keep
[root@machine netdata]#
```

**Comment by ktsaou** [comment]
It seems I missed this issue.
If anyone can help, please do so...
Otherwise, I will install CentOS 7 on a VM to test this.

**Comment by samueljon** [comment]
I'm going to try to change the security context on the log folder and see if that helps and report back. 

```sh
semanage fcontext -a -t var_log_t "/opt/netdata/var/log/netdata(/.*)?"
restorecon -v /opt/netdata/var/log/netdata 
```

**Comment by samueljon** [comment]
I suggest that logrotate is changed to rotate logs as netdata user.

```CONF
/opt/netdata/var/log/netdata/*.log {
	su netdata netdata # ADD THIS
	daily
	missingok
	rotate 14
	compress
	delaycompress
	notifempty
	sharedscripts
	postrotate
		/bin/kill -HUP `pidof netdata 2>/dev/null` 2>/dev/null || true
	endscript
}
```

And when selinux is used update the  security context.
```shell
semanage fcontext -a -t var_log_t "/opt/netdata/var/log/netdata(/.*)?"
restorecon -v /opt/netdata/var/log/netdata 
```

**Comment by klatiss** [comment]
I had this same issue which was fixed simply with applying the SELinux policy above. I did not need to change the logrotate configuration from the default.

**Comment by paulfantom** [comment]
I can confirm that this is due to wrong SELinux context.

**Comment by efes0** [comment]
In RHEL based distros `/var/log/.*` has a SElinux file_type of `var_log_t` so when using a default git install the netdata log folder will inherit that file context, however when installing the static build or using the `--install` option it places the log folder elsewhere and it then inherits the wrong file_type context blocking logrotate from accessing the log files and requiring the fix from https://github.com/netdata/netdata/issues/2794#issuecomment-334386546 (should also use `-r` option with that).

Also if i understand [workaround SELinux, fix #4 #6](https://github.com/romain-dartigues/ansible-netdata/pull/6) correctly that won't work if someone uses the `--install` option and installs it somewhere other then `/opt`. Ideally you'd want to read the path passed to `--install` and take it from there.
